pkg/api/steve/proxy/path.go

```diff
@@ -0,0 +1,70 @@
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var errBadResource = errors.New("unidentified resource")
+var errBadGVR = errors.New("invalid GVR")
+
+// gvrFromPath tries to parse a URL path and extract GVR information from it.
+// Version must be present.
+// Group may be empty for core discovery client paths.
+// Resource may be empty for non-core discovery client paths.
+// This won't handle paths where "namespaces" designates a custom Namespace kind. So this function is not generally
+// applicable for any paths including any resources.
+func gvrFromPath(path string) (*schema.GroupVersionResource, error) {
+	parts := pathParts(path)
+	// Allow paths for core resource discovery, such as "/api/v1". And disallow non-core short paths, such as "/apis/management.cattle.io".
```

Comment:
Nit: I don't think this comment should be dealing in allow/disallow speak. This function just extracts a GVR, it doesn't make any allow/deny calls.

Reply:
Agreed, will adjust.